### DISCUSSION TITLE
Set up HTTPS

### DISCUSSION THREAD

**Discussion by ashears** [fastapi/fastapi#7927] [Questions] [answered]
**Description**
I have been stuck on how to set up HTTPS.
I have the certificate I would like to point to.

How can I configure my app to use this certificate for HTTPS?
Any examples would be greatly appreciated!

**Comment by ashears** [comment]
From researching Starlette, it seems it is the _middleware_ that would achieve this.
Will research more and close this if I’m right.

**Comment by ashears** [comment]
I take that back, middleware does not seem to take in a certificate location.

**Comment by ashears** [comment]
I think it is in Gunicorn that takes in the file.

**Comment by dmontagu** [comment, marked as answer]
Gunicorn may be able to handle this, but I would also recommend looking at using a reverse proxy to handle the ssl stuff. You can look at https://dockerswarm.rocks/traefik/ for a guide that shows how to use traefik to enable ssl. This is probably the easiest way to get https support with the full stack fastapi templates, for example.

**Comment by ashears** [comment]
Thanks @dmontagu !

I think I understand enough to be able to continue to work on this, and will close this question.
:)

**Comment by tiangolo** [comment]
Thanks for the help here @dmontagu ! :clap: :bow: 

Thanks for reporting back and closing the issue @ashears  :+1:
